### DISCUSSION TITLE
next.lock / lock.json file fails to automatically generate

### DISCUSSION THREAD

**Discussion by dmfcoops** [vercel/next.js#32698] [Help]
When a new/updated data source is imported into an API and I run **npm run dev** or **npx next dev**  the lock.json file is not automatically updated.

So, when I run a Production build using **npm run build** or **npx next build** I receive an error:

**Module not found: <datasource> has no lockfile entry and lockfile is frozen**

How can I force Next to regenerate the lock.json file using the source of the new import?

**Comment by idleberg** [comment]
Does that module producing the error have an `urlImport`?
